### PR TITLE
fix(semver): Ranges can fail with some pre-release numbering

### PR DESCRIPTION
**Pull request by mattfarina**
There was a hyphen range bug in semver that prevented constraints
in the form of v2.3.5-20161202202307-sha.e8fc5e5. This would be
parsed incorrectly as a hyphen range and then fail. The bug in
semver was fixed in semver v1.2.2 for this issue.

Note, the only package updated was Masterminds/semver.

### CODE REVIEW

**Comment by adamreese** [comment]
@mattfarina looks good but needs a rebase

**Comment by mattfarina** [comment]
Rebased and fixed missing test dependencies

File: glide.lock [hunk 1]
```diff
@@ -490,4 +490,14 @@ imports:
   - third_party/forked/golang/netutil
   - third_party/forked/golang/reflect
   - third_party/forked/golang/template
-testImports: []
+testImports:
```
Comment (adamreese) [inline glide.lock]: Should these also be in `glide.yaml`?

Comment (mattfarina) [inline glide.lock]: @adamreese nope. these test imports are dependencies of dependencies of helm. No need for them in the `glide.yaml` as they are transitive.

**Pull request merged**